lua/neotest/client/strategies/dap/init.lua

```diff
@@ -65,14 +63,10 @@ return function(spec)
       return result_code ~= nil
     end,
     output_stream = function()
-      local sender, receiver = async.control.channel.mpsc()
-      data_accum:subscribe(function(d)
-        sender.send(d)
-      end)
+      local queue = nio.control.queue()
```

Comment:
Wow, this pr  is a huge change. Actually, I never quite understood the async feature in Lua before. The new API looks much easier to understand.

By the way, I don't know if you've seen Elixir's [Task.async](https://hexdocs.pm/elixir/1.14.4/Task.html) module. I've always thought that the process model of Elixir/Erlang is the most understandable concurrency model.

Reply:
I'm not familiar with Elixir but from a quick look it looks quite similar to Python's `asyncio` and thus this library with the `Task` object already :smile: Is there anything in particular that you think would be worth adapting from Elixir?

Reply:
Nope, I'm not familiar with Lua and Python asynchronous. I just said it casually, but the functions they implemented should be similar, although the underlying technology is probably very different.